addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/LEDStateDTO.java

```diff
@@ -70,18 +75,18 @@ public static LEDStateDTO valueOf(int state, int program, int programSpeed, int
         OnOffType power = (state & 0x01) != 0 ? OnOffType.ON : OnOffType.OFF;
         float[] hsv = new float[3];
         Color.RGBtoHSB(red, green, blue, hsv);
-        long hue = (long) (hsv[0] * 360);
-        int saturation = (int) (hsv[1] * 100);
-        int brightness = (int) (hsv[2] * 100);
+        BigDecimal hue = BigDecimal.valueOf(hsv[0] * 360);
+        BigDecimal saturation = BigDecimal.valueOf (hsv[1] * 100);
+        BigDecimal brightness = BigDecimal.valueOf (hsv[2] * 100);
         DecimalType h = new DecimalType(hue);
         PercentType s = new PercentType(saturation);
         // set Brightness to 0 if state is OFF
         PercentType b = new PercentType(brightness);
-        PercentType w = new PercentType(white / 255 * 100);
-        PercentType w2 = new PercentType(white2 / 255 * 100);
+        PercentType w = new PercentType(BigDecimal.valueOf(Math.round(((double) white / 255) * 100)));
```

Comment:
Flor clarity I would prefer to leave away the `BigDecimal.valueOf`, because you actually pass the int value and use a double calculation, passing the int will and makes the intent more clear. That percentType uses a `BigDecimal` as internal storage is imho an implementation details and no reason to use it here.